content/blog/2019-01-28-subqueries.md

```diff
@@ -0,0 +1,59 @@
+
+With subquery support, all the waiting and frustration is taken care of.
+
+## Subqueries
+
+A subquery is similar to a [/api/v1/query_range](https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries) API call, but embedded within an instant query. The result of a subquery is a range vector.
+
+The Prometheus team arrived at a consensus for the syntax of subqueries at the Prometheus Dev Summit 2018 held in Munich. These are the [notes of the summit on subquery support](https://docs.google.com/document/d/1-C5PycocOZEVIPrmM1hn8fBelShqtqiAmFptoG4yK70/edit#heading=h.q32gdnoqz8t0), and a brief [design doc for the syntax](https://docs.google.com/document/d/1P_G87zN88YvmMr4iwLWygChMTZhai1L7S_c0awu1CAE/edit?usp=sharing) used for implementing subquery support.
+
+    <instant_query> '[' <range> ':' [ <resolution> ] ']' [ offset <duration> ]
+
+* `<instant_query>` is equivalent to `query` field in `/query_range` API.
+* `<range>` and `offset <duration>` is similar to a range selector.
+* `<resolution>` is optional, which is equivalent to `step` in `/query_range` API.
+
+When the resolution is not specified, the global evaluation interval is taken as the default resolution for the subquery. Also, the step of the subquery is aligned independently, and does not depend on the parent query's evaluation time.
+
+## Examples
+
+The subquery inside the `min_over_time` function returns the 5-minute rate of the `http_requests_total` metric for the past 30 minutes, at a resolution of 1 minute. This would be equivalent to a `/query_range` API call with `query=rate(http_requests_total[5m]), end=<now>, start=<now>-30m, step=1m`, and taking the min of all received values.
+
+    min_over_time( rate(http_requests_total[5m])[30m:1m] )
+
+Breakdown
+
+* `rate(http_requests_total[5m])[30m:1m]` is the subquery, where `rate(http_requests_total[5m])` is the query to be executed.
+* `rate(http_requests_total[5m])` is executed from `start=<now>-30m` to `end=<now>`, at a resolution of `1m`. Note that `start` time is aligned independantly with step of `1m` (aligned steps are `0m 1m 2m 3m ...`).
+* Finally the result of all the evaluations above are passed to `min_over_time()`.
+
+Below is an example of a nested subquery, and usage of default resolution. The innermost subquery gets the rate of `distance_covered_meters_total` over a range of time. We use that to get `deriv()` of the rates, again for a range of time. And finally take the max of all the derivatives.
+Note that the `<now>` time for the innermost subquery is relative to the evaluation time of the outer subquery on `deriv()`.
+
+    max_over_time( deriv( rate(distance_covered_meters_total[1m])[5m:1m] )[10m:] )
+
+In most cases you would require the default evaluation interval, which is the interval at which rules are evaluated by default. Custom resolution will be helpful in cases where you want to compute less/more frequently, e.g. expensive queries which you might want to compute less frequently.
```

Comment:
Either "A custom resolution" or "Custom resolutions" (the singular of a noun without an article only works in few scenarios)